app/scripts/views/settings.js

```diff
@@ -88,29 +233,69 @@ function (Cocktail, Session, FormView, BaseView, AvatarMixin,
              this._able.choose('avatarLinkVisible', { email: email });
     },
 
-    _areCommunicationPrefsVisible: function () {
-      return this._able.choose('communicationPrefsVisible', {
-        lang: this.navigator.language
-      });
-    },
-
     _setupAvatarChangeLinks: function (show) {
       if (show) {
-        this.$('.change-avatar-text').css('visibility', 'visible');
         this.$('.avatar-wrapper > *').wrap('<a href="/settings/avatar/change" class="change-avatar"></a>');
       }
     },
 
-    afterVisible: function () {
+    _showAvatar: function () {
       var self = this;
       var account = self.getSignedInAccount();
-
-      FormView.prototype.afterVisible.call(self);
       return self.displayAccountProfileImage(account)
         .then(function () {
           self._setupAvatarChangeLinks(self._isAvatarLinkVisible(account));
         });
+    },
+
+    _areCommunicationPrefsVisible: function () {
+      return !! this._able.choose('communicationPrefsVisible', {
+        lang: this.navigator.language
+      });
+    },
+
+    signOut: allowOnlyOneSubmit(function () {
+      var self = this;
+      var sessionToken = self.getSignedInAccount().get('sessionToken');
+
+      self.logScreenEvent('signout.submit');
+      return self.fxaClient.signOut(sessionToken)
+        .fail(function () {
+          // ignore the error.
+          // Clear the session, even on failure. Everything is A-OK.
+          // See issue #616
+          self.logScreenEvent('signout.error');
+        })
+        .fin(function () {
+          self.logScreenEvent('signout.success');
+          self.user.clearSignedInAccount();
+          Session.clear();
+          self.navigate('signin', {
+            success: t('Signed out successfully')
+          });
+        });
+    }),
+
+    SUCCESS_MESSAGE_DELAY: 3000, // show success message for 3 seconds
+
+    displaySuccess: function () {
+      var self = this;
+      clearTimeout(self._successTimeout);
```

Comment:
Let's split it out when we need to– perhaps there will be a better solution when the time comes and the requirements are clearer.